packages/flutter/lib/src/material/bottom_navigation_bar.dart

```diff
@@ -373,8 +404,20 @@ class _BottomNavigationTile extends StatelessWidget {
 
     final double selectedFontSize = selectedLabelStyle.fontSize;
 
-    double bottomPadding = selectedFontSize / 2.0;
-    double topPadding = selectedFontSize / 2.0;
+
+    final double selectedIconSize = selectedIconTheme?.size ?? iconSize;
+    final double unselectedIconSize = unselectedIconTheme?.size ?? iconSize;
+    final double selectedIconDiff = math.max(selectedIconSize - unselectedIconSize, 0);
+    final double unselectedIconDiff = math.max(unselectedIconSize - selectedIconSize, 0);
+
+    double bottomPadding = Tween<double>(
+      begin: selectedFontSize / 2.0 + selectedIconDiff / 2.0,
+      end: selectedFontSize / 2.0 + unselectedIconDiff / 2.0,
```

Comment:
Is this supposed to be `unselectedFontSize`?

Reply:
Good question, both of these *are* actually meant to be `selectedFontSize`. We have to use the same variable for both, because all of the `BottomNavigationBarItems` should always be the same height. Otherwise we get an unintended animation.